src/test/java/ir/sahab/kafkaconsumer/OffsetTrackerTest.java

```diff
@@ -100,6 +100,81 @@ public void testDisorderedAckPages() {
         Assert.assertEquals(4, offsetToCommit.getAsLong());
     }
 
+    @Test
+    public void testPagesWithGap() {
+        final int pageSize = 3;
+        final int maxOpenPagesPerPartition = 2;
+        final OffsetTracker offsetTracker = new OffsetTracker(pageSize, maxOpenPagesPerPartition);
+        final int partition = 0;
+
+        // Track calls which opens the first page: [0..2]
+        offsetTracker.track(partition, 1);
+
+        // Track calls which opens the second page: [3..5]
+        offsetTracker.track(partition, 3);
+
+
```

Comment:
Add comment:
// Offset 1 completes the first page because the next offsets are inside the recognized gap.